### PR TITLE
[avatar] Fix image status edge cases

### PR DESCRIPTION
**Pull request by atomiks**
This is part of the Codex component behavior and test coverage sweep. It fixes Avatar behavior/API issues found during review and adds focused regression coverage for the affected paths.

## Root cause

Avatar had edge cases where its implementation and tests did not fully cover the expected image loading, fallback, and native image prop contract.

## Changes

- Fix Avatar image prop forwarding for `crossOrigin`, `referrerPolicy`, `srcSet`, and `sizes`.
- Use the same responsive image inputs for the loading probe that the rendered image receives.
- Reset image loading status when `Avatar.Image` unmounts so fallback can render again.
- Reconcile `Avatar.Fallback` delay changes.
- Add regression and type tests for the affected Avatar paths.

## Original findings addressed

- [P2] `Avatar.Image` drops `crossOrigin` and `referrerPolicy` from the rendered ``.
- [P2] `srcSet`/`sizes` are part of the native image API but loading status only looks at `src`.
- [P2] Removing a loaded `Avatar.Image` can leave fallback hidden forever.
- [P3] `Fallback.delay` is one-shot state and does not reconcile prop changes.

### CODE REVIEW

File: packages/react/src/avatar/fallback/AvatarFallback.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const AvatarFallback = React.forwardRef(function AvatarFallback(
   const { className, render, delay, style, ...elementProps } = componentProps;
 
   const { imageLoadingStatus } = useAvatarRootContext();
-  const [delayPassed, setDelayPassed] = React.useState(delay === undefined);
+  const [delayPassed, setDelayPassed] = React.useState(false);
   const timeout = useTimeout();
 
   React.useEffect(() => {
```
[the remark this answers is not in the document]
Comment (atomiks) [reply packages/react/src/avatar/fallback/AvatarFallback.tsx]: Indeed `delay={100}` → `delay={200}` after the first delay elapsed won't re-initiate. That's intentional though: delay exists to prevent fallback flash during fast image loads. Once the fallback is already visible, re-applying a delay would mean hiding rendered content and showing it again later (the reset would happen in an effect after commit, so it'd paint → vanish → reappear). There's no real scenario where re-hiding a visible fallback on a delay change is desirable.

The case that mattered (delay becoming undefined while the fallback was stuck hidden) is handled at render time without flicker.

**Review by michaldudak — COMMENTED** [review]
# PR review

No merge-blocking issues found. The focused Avatar jsdom suites pass, but there is one small regression in dynamic `delay` behavior.

## Bugs (1)

### 1. 🟡 Preserve visible fallback when delay is added after mount

**Location:** `packages/react/src/avatar/fallback/AvatarFallback.tsx:23`

```tsx
const [delayPassed, setDelayPassed] = React.useState(false);

// ...

enabled: imageLoadingStatus !== 'loaded' && (delay === undefined || delayPassed),
```

When `Avatar.Fallback` starts without a `delay`, the fallback is visible immediately because `delay === undefined`. If the prop later changes to a number, `delayPassed` is still `false`, so the fallback disappears until the timer elapses. On `master`, `delayPassed` initialized to `delay === undefined`, so this transition kept the already-visible fallback mounted.

**Failure scenario:** A controlled avatar renders fallback immediately, then a settings/state update changes `delay` from `undefined` to `600`; the initials disappear for 600ms even though the image is still missing or errored.

**Fix:** Initialize `delayPassed` with `delay === undefined`, and add the inverse regression test for `undefined -> number`.

## Tests (0)

No findings.

## Simplifications (0)

No findings.

## Docs (0)

No findings.

## Verdict

**Approve after nits** — one non-blocking dynamic prop regression should be fixed.

File: packages/react/src/avatar/fallback/AvatarFallback.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const AvatarFallback = React.forwardRef(function AvatarFallback(
     ref: forwardedRef,
     props: elementProps,
     stateAttributesMapping: avatarStateAttributesMapping,
-    enabled: imageLoadingStatus !== 'loaded' && delayPassed,
+    enabled: imageLoadingStatus !== 'loaded' && (delay === undefined || delayPassed),
```
Comment (michaldudak) [inline packages/react/src/avatar/fallback/AvatarFallback.tsx:46]: 🟡 Delay can hide fallback again after it was shown

When `delay` changes from a number to `undefined`, the fallback becomes visible because of the new `delay === undefined` branch, but `delayPassed` remains `false`. If `delay` is later set back to a number, the fallback hides again until the new timer completes.

Failure scenario: an errored avatar starts with `delay={100}`, changes to `delay={undefined}` before the timer fires, then changes back to `delay={100}`. The fallback flashes visible, then disappears again even though it had already been shown.

Fix: when `delay === undefined`, set `delayPassed` to `true` in the effect, and add a regression test for `100 -> undefined -> 100`.

**Review by michaldudak — COMMENTED** [review]
I found one non-blocking delay reconciliation bug. CI is green, and the rest of the Avatar image/status changes looked okay in this pass.

**Review by michaldudak — APPROVED** [review]
The last issue Codex found looks like a very unlikely edge case. No need to fix it IMO if it increases the bundle size.
